functions/go/fix/fixpkg/fix.go

```diff
@@ -240,38 +250,87 @@ func (s *Fix) FunctionsInPkg(nodes []*yaml.RNode, pkgPath string) []v1alpha2.Fun
 	return res
 }
 
-// FixKptfile migrates the input Kptfile node from v1alpha1 to v1alpha2
-func (s *Fix) FixKptfile(node *yaml.RNode, functions []v1alpha2.Function) (*yaml.RNode, error) {
+// FixKptfile migrates the input Kptfile node from v1alpha1 or v1alpha2 to v1
+func (s *Fix) FixKptfile(node *yaml.RNode, functions []v1.Function) (*yaml.RNode, error) {
 	var err error
 	meta, err := node.GetMeta()
 	if err != nil {
 		return node, err
 	}
 
+	settersConfigFilePath := filepath.Join(filepath.Dir(meta.Annotations[kioutil.PathAnnotation]), SettersConfigFileName)
+
+	// v1alpha2 to v1 migration
+	if strings.Contains(meta.APIVersion, "v1alpha2") {
+		node.SetApiVersion(v1.KptFileAPIVersion)
+		s.Results = append(s.Results, &Result{
+			FilePath: meta.Annotations[kioutil.PathAnnotation],
+			Message:  fmt.Sprintf("Updated apiVersion to %s", v1.KptFileAPIVersion),
+		})
+
+		kf, err := v1.ReadFile(node)
+		if err != nil {
+			return node, err
+		}
+
+		// apply-setters function input should be moved to configPath option as it the
+		// best practice
+		if kf.Pipeline != nil {
+			for i, fn := range kf.Pipeline.Mutators {
+				if strings.Contains(fn.Image, "apply-setters:v0.1") && len(fn.ConfigMap) > 0 {
+					settersConfig, err := ConfigFromSetters(kf.Pipeline.Mutators[0].ConfigMap, settersConfigFilePath)
+					if err != nil {
+						return node, err
+					}
+					s.settersConfigs = append(s.settersConfigs, settersConfig)
+					kf.Pipeline.Mutators[i].ConfigMap = nil
+					kf.Pipeline.Mutators[i].ConfigPath = SettersConfigFileName
+					s.Results = append(s.Results, &Result{
+						FilePath: settersConfigFilePath,
+						Message:  `Moved setters from configMap to configPath`,
+					})
+				}
+			}
+		}
+
+		// convert updated kf to yaml node
+		b, err := yaml.Marshal(kf)
```

Comment:
do we have to set encoding etc to take care of formatting ?

Reply:
We don't need to. It is not writing to disk. It is parsing it back to RNode, so the formatting doesn't matter.

Reply:
ok, got it.